service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionPatronRequestISO18626StatusChangeService.groovy

```diff
@@ -16,13 +17,19 @@ import org.olf.rs.statemodel.StateModel
  */
 public class ActionPatronRequestISO18626StatusChangeService extends ActionISO18626RequesterService {
 
+    private static final String STATUS_UNFILLED = "Unfilled"
+
     @Override
     String name() {
         return(ReasonForMessage.MESSAGE_REASON_STATUS_CHANGE)
     }
 
     @Override
     ActionResultDetails performAction(PatronRequest request, Object parameters, ActionResultDetails actionResultDetails) {
+        if (EventMessageRequestIndService.isSlnpRequesterStateModel(request) && parameters.statusInfo?.status == STATUS_UNFILLED) {
+            request.state = request.stateModel.initialState
```

Comment:
@JanisSaldabols did you test that this works? Is this simpler than doing it in`ActionPatronRequestISO18626StatusChangeService`?

Reply:
This is ActionPatronRequestISO18626StatusChangeService class